modules/admin-ui-picard/app/src/components/users/partials/modal/UserDetails.js

```diff
@@ -0,0 +1,118 @@
+
+    const initialValues = {
+        ...userDetails,
+        password: ''
+    };
+
+    // information about tabs
+    const tabs = [
+        {
+            tabNameTranslation: 'USERS.USERS.DETAILS.TABS.USER',
+            name: 'general'
+        },
+        {
+            tabNameTranslation: 'USERS.USERS.DETAILS.TABS.ROLES',
+            name: 'roles'
+        },
+        {
+            tabNameTranslation: 'USERS.USERS.DETAILS.TABS.EFFECTIVEROLES',
+            name: 'effectiveRoles'
+        }
+    ];
+
+    const openTab = tabNr => {
+        setPage(tabNr);
+    };
+
+    const handleSubmit = values => {
+        updateUserDetails(values, userDetails.username);
+        close();
+    }
+
+    return (
+        <>
+            {/* Navigation */}
+            <nav className="modal-nav" id="modal-nav">
+                <a className={cn({active: page === 0})}
+                   onClick={() => openTab(0)}>
+                    {t(tabs[0].tabNameTranslation)}
+                </a>
+                <a className={cn({active: page === 1})}
+                   onClick={() => openTab(1)}>
+                    {t(tabs[1].tabNameTranslation)}
+                </a>
+                <a className={cn({active: page === 2})}
+                   onClick={() => openTab(2)}>
+                    {t(tabs[2].tabNameTranslation)}
+                </a>
+            </nav>
+
+            {/* formik form used in entire modal */}
+            <Formik initialValues={initialValues}
+                    validationSchema={EditUserSchema}
+                    onSubmit={values => handleSubmit(values)}>
+                {formik => (
+                    <>
+                        {page === 0 && (
+                            <EditUserGeneralTab formik={formik}/>
+                        )}
+                        {page === 1 && (
+                            <UserRolesTab formik={formik}/>
+                        )}
+                        {page === 2 && (
+                            <UserEffectiveRolesTab formik={formik}/>
+                        )}
+
+                        {/* Navigation buttons and validation */}
+                        <footer>
+                            <button className={cn("submit", {
+                                        active: (formik.dirty && formik.isValid),
+                                        inactive: !(formik.dirty && formik.isValid)})
+                                    }
+                                    disabled={!(formik.dirty && formik.isValid)}
+                                    onClick={() => formik.handleSubmit()}
+                                    type="submit">
+                                {t('SUBMIT')}
+                            </button>
+                            <button className="cancel"
+                                    onClick={() => close()}>
+                                {t('CANCEL')}
+                            </button>
+                        </footer>
```

Comment:
This footer is also shown on `UserEffectiveRolesTab` where I think the user can never make any input that could be saved. I think it would be a good idea, to not display the footer for that tab.